cxx/fbjni/detail/Environment.cpp

```diff
@@ -93,7 +94,8 @@ JNIEnv* attachCurrentThread() {
 
   const auto threadName = getThreadName();
   if (threadName.size()) {
-    args.name = threadName.c_str();
+    std::vector<char> v(threadName.begin(), threadName.end());
```

Comment:
vector is not guaranteed to be null-terminated, and being in this sub scope means it's immediately destroyed. Why was `.c_str()` not working?

Reply:
Good catch. The other PR has a better fix in place. This is needed because const qualifier is being dropped. The code in the main branch is currently broken.

Reply:
Yeah, this was my fault, but I caught it before landing.